hub/src/main/java/org/motechproject/hub/web/HubController.java

```diff
@@ -53,37 +56,84 @@
 
     private SettingsFacade settingsFacade;
 
+    /**
+     * Creates an instance of HubController using autowired SettingsFacade instance.
+     *
+     * @param settingsFacade autowired SettingsFacade settings
+     */
     @Autowired
     public HubController(
             @Qualifier("hubSettings") final SettingsFacade settingsFacade) {
         this.settingsFacade = settingsFacade;
     }
 
+    /**
+     * Gets the {@link org.motechproject.hub.validation.HubValidator} responsible
+     * for request input parameters validation.
+     *
+     * @return the hub validator
+     */
     public HubValidator getHubValidator() {
         return hubValidator;
     }
 
+    /**
+     * Sets the {@link org.motechproject.hub.validation.HubValidator} responsible
+     * for request input parameters validation.
+     *
+     * @param hubValidator <code>HubValidator</code> to be set
+     */
     public void setHubValidator(HubValidator hubValidator) {
         this.hubValidator = hubValidator;
     }
 
+    /**
+     * Gets the <code>SubscriptionService</code> responsible for performing
+     * subscribe and unsubscribe actions.
+     *
+     * @return the subscription service
+     */
     public SubscriptionService getSubscriptionService() {
         return subscriptionService;
     }
 
+    /**
+     * Sets the subscription service responsible for performing
+     * subscribe and unsubscribe actions.
+     *
+     * @param subscriptionService the <code>SubscriptionService</code> to be set
+     */
     public void setSubscriptionService(SubscriptionService subscriptionService) {
         this.subscriptionService = subscriptionService;
     }
 
+    /**
+     * Gets the <code>ContentDistributionService</code> responsible for distributing
+     * content to subscribers.
+     *
+     * @return the content distribution service
+     */
     public ContentDistributionService getContentDistributionService() {
         return contentDistributionService;
     }
 
+    /**
+     * Sets the content distribution service responsible for distributing
+     * content to subscribers.
+     *
+     * @param contentDistributionService the <code>ContentDistributionService</code> to be set
+     */
     public void setContentDistributionService(
             ContentDistributionService contentDistributionService) {
         this.contentDistributionService = contentDistributionService;
     }
 
+    /**
+     * Gets hub settings, which at the moment contains only hub base URL.
```

Comment:
don't mention what they contain here, since we will have to update this javadoc as well if decide to extend the settings